src/display.c

```diff
@@ -0,0 +1,233 @@
+
+	if (display_dev == NULL) {
+		printk("device not found.  Aborting test.");
+		return -1;
+	}
+
+	display_top_bar = lv_cont_create(lv_scr_act(), NULL);
+	lv_obj_set_width(display_top_bar, lv_obj_get_width(lv_scr_act()));
+	lv_cont_set_layout(display_top_bar, LV_LAYOUT_PRETTY_TOP);
+
+	display_bot_bar = lv_cont_create(lv_scr_act(), NULL);
+	lv_obj_set_width(display_bot_bar, lv_obj_get_width(lv_scr_act()));
+	lv_cont_set_layout(display_bot_bar, LV_LAYOUT_PRETTY_BOTTOM);
+
+	display_center_pane = lv_cont_create(lv_scr_act(), NULL);
+	lv_obj_align(display_center_pane, display_top_bar, LV_ALIGN_OUT_BOTTOM_LEFT, 0, 0);
+	lv_obj_set_width(display_center_pane, lv_obj_get_width(lv_scr_act()));
+	lv_obj_set_height(display_center_pane, lv_obj_get_height(lv_scr_act()) - lv_obj_get_height(display_top_bar) - lv_obj_get_height(display_bot_bar));
+	lv_cont_set_layout(display_center_pane, LV_LAYOUT_COLUMN_MID);
+
+	lv_obj_align(display_bot_bar, display_center_pane, LV_ALIGN_OUT_BOTTOM_LEFT, 0, 0);
+
+	display_contacts_label = lv_label_create(display_center_pane, NULL);
+	lv_obj_set_height_margin(display_contacts_label, 20);
+	lv_label_set_text(display_contacts_label, "");
+
+	if (IS_ENABLED(CONFIG_LVGL_POINTER_KSCAN)) {
+		risk_contacts_button = lv_btn_create(display_center_pane, NULL);
+		lv_btn_set_fit(risk_contacts_button, LV_FIT_TIGHT);
+		display_risk_contacts_label = lv_label_create(risk_contacts_button, NULL);
+	} else {
+		display_risk_contacts_label = lv_label_create(display_center_pane, NULL);
+	}
+
+	lv_obj_set_height_margin(display_risk_contacts_label, 20);
+	lv_label_set_text(display_risk_contacts_label, "");
+
+    display_memory_label = lv_label_create(display_top_bar, NULL);
+	lv_obj_set_width_margin(display_memory_label, 10);
+	lv_label_set_text(display_memory_label, "");
+
+	display_clock_label = lv_label_create(display_top_bar, NULL);
+	lv_obj_set_width_margin(display_clock_label, 10);
+	lv_label_set_text(display_clock_label, "");
+
+	display_battery_label = lv_label_create(display_top_bar, NULL);
+	lv_obj_set_width_margin(display_battery_label, 10);
+	lv_label_set_text(display_battery_label, "");
+
+	display_msg_label = lv_label_create(display_bot_bar, NULL);
+	lv_label_set_text(display_msg_label, "");
+	
+	update_display();
+
+	lv_task_handler();
+	display_blanking_off(display_dev);
+	#endif
+
+	return 0;
+}
+
+int display_set_message(char* msg) {
+	#ifdef DISPLAY
+	lv_label_set_text(display_msg_label, msg);
+	#endif
+	return 0;
+}
+
+int display_set_time(int time) {
+	return 0;
+}
+
+int display_set_bat(int bat) {
+	#ifdef DISPLAY
+	lv_label_set_text_fmt(display_battery_label, "Bat: %d%%", bat);
+	#endif
+	return 0;
+}
+
+int display_set_mem(int mem) {
+	#ifdef DISPLAY
+	lv_label_set_text_fmt(display_memory_label, "Mem: %d%%", mem);
+	#endif
+	return 0;
+}
+
+int display_set_contacts(int contacts) {
+	#ifdef DISPLAY
+	if (contacts == 1) {
+		lv_label_set_text_fmt(display_contacts_label, "%d Kontakt erkannt, davon", contacts);
```

Comment:
Please use English also for the UI texts (for now). Please also try to use simple statements as "Number of contacts: %d" to prevent case-dependent translations.